core/src/mast/serialization/basic_block_data_decoder.rs

```diff
@@ -139,6 +139,22 @@ impl<'a> BasicBlockDataDecoder<'a> {
                 let num_cycles = data_reader.read_u8()?;
                 let should_break = data_reader.read_bool()?;
 
+                // source location
+                let location = if data_reader.read_bool()? {
+                    let str_index_in_table = data_reader.read_usize()?;
+                    let source_file =
+                        crate::SourceFile::from(self.read_string(str_index_in_table)?);
```

Comment:
Would this allocate a new string for every operation?

Even if so, I think it is probably fine for now - but we should create an issue to make it more efficient in the future. For example, `MastForest` could also have an optional string table and these could be indexes into that table (I think you also mention something like this in the PR description).

Another option is to enable optional serialization of the source MASM into the resulting `MastForest`. This way, references to source files won't need to be filesystem paths but could rather be `LibraryPaths`.